compiler-core/src/warning.rs

```diff
@@ -186,6 +188,15 @@ impl Warning {
                 }
             },
 
+            Warning::InvalidSource { name } => Diagnostic {
+                title: "Invalid module name. Module names must begin with a lowercase letter and contain only lowercase alphanumeric characters or underscores.".into(),
```

Comment:
This should be text rather than the title. The title is short and more detail goes in the text field, wrapped at around 65 characters to make it read nicely.